app/src/test/java/com/android/sample/model/receivers/NotificationReceiverTest.kt

```diff
@@ -0,0 +1,92 @@
+package com.android.sample.receivers
+
+import android.app.NotificationManager
+import android.content.Context
+import android.content.Intent
+import android.content.res.Resources
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import org.junit.Before
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.mockito.kotlin.any
+import org.mockito.kotlin.mock
+import org.mockito.kotlin.never
+import org.mockito.kotlin.verify
+import org.mockito.kotlin.whenever
+
+@RunWith(AndroidJUnit4::class)
+class NotificationReceiverTest {
```

Comment:
All the "not working" tests cases are tested very good! And the case where the notification is created is in the android test, these can be unified but it is not a big matter